### ISSUE TITLE
Redis client connection leak when using Pub/Sub

### ISSUE THREAD

**Issue by Eng-Fouad** [quarkusio/quarkus#28656]
### Describe the bug

There is a connection leak when connecting to Redis using `subscribe` then `unsubscribe`. The connection is never closed, and a new connection is established upon using `subscribe` again. Eventually, Redis's `maxclients` is reached, and we cannot use `subscribe` anymore.

I confirmed that using `redis-cli -> info clients`.

I think it is related to a bug in vert.x Redis client: [PooledRedisConnection.close() does not close the connection if it is tainted (Pub/Sub)](https://github.com/vert-x3/vertx-redis-client/issues/353)

### Output of `java -version`

17

### GraalVM version (if different from Java)

22.2

### Quarkus version or git rev

2.13.2.Final

**Comment by cescoffier** [comment]
Do you have a reproducer?
I just tried:

```java
        Person p = new Person("a", "b");
        for (int i = 0; i < 100; i++) {
            System.out.println("iteration " + i);
            AtomicReference<Person> received = new AtomicReference<>();
            var subscriber = pubsub.subscribe("some-channel", received::set);
            pubsub.publish("some-channel", p);
            Awaitility.await().until(() -> received.get() != null);
            subscriber.unsubscribe("some-channel");
        }
```

Works perfectly.

**Comment by Eng-Fouad** [comment]
@cescoffier Here is a reproducer: https://github.com/Eng-Fouad/quarkus-redis-connection-leak

**Comment by cescoffier** [comment]
Thanks, is it possible to have something simpler? It's hard to follow what's going on, and some connection seems to not be closed.

**Comment by cescoffier** [comment]
Got it! I've simplified your code. I have a test (I didn't know that max client was configured to 10 000)

**Comment by cescoffier** [comment]
I confirm, it's https://github.com/vert-x3/vertx-redis-client/issues/353.

**Comment by Eng-Fouad** [comment]
@cescoffier vetx-redis-client 4.3.5 is released, which has the fix for this issue.

**Comment by cescoffier** [comment]
Yes, I know. We will update vertx in Quarkus 2.15 (December)

**Comment by cescoffier** [comment]
Fixed.
